darts/models/filtering/kalman_filter.py

```diff
@@ -197,48 +204,50 @@ def filter(
         y_values = series.values(copy=False)
         if self._expect_covariates:
             u_values = covariates.values(copy=False)
+
+            # set control signal to 0 if it contains NaNs:
+            u_values = np.nan_to_num(u_values, copy=True, nan=0.0)
         else:
             u_values = np.zeros((len(y_values), 0))
 
         # For each time step, we'll sample "n_samples" from a multivariate Gaussian
         # whose mean vector and covariance matrix come from the Kalman filter.
         if num_samples == 1:
-            sampled_states = np.zeros(
+            sampled_outputs = np.zeros(
                 (
                     len(y_values),
                     self.dim_y,
                 )
             )
         else:
-            sampled_states = np.zeros((len(y_values), self.dim_y, num_samples))
+            sampled_outputs = np.zeros((len(y_values), self.dim_y, num_samples))
 
+        mean_vec = None  # store previous mean
         for i in range(len(y_values)):
             y = y_values[i, :].reshape(-1, 1)
+
+            if np.isnan(y).any():
+                raise_if(i == 0, "The first value of the series must not be NaN.")
+                # if an observation is missing, take the previous mean as an observation
+                y = mean_vec.reshape(-1, 1)
```

Comment:
I'm not sure if this is the best way of handling missing values here, especially if used for forecasting. I think this might result in an overconfident probabilistic output. I will have to take a closer look, but maybe we should use `nfoursid`'s `extrapolate()` method instead or add something similar for this purpose?

Reply:
You're right, this is not quite correct. I guess we should run the state space system one step instead. I have [opened an issue with nfoursid](https://github.com/spmvg/nfoursid/issues/3) to ask what might be the best way to do it.

Reply:
@tomasvanpottelbergh I have now changed it to take a step of the Kalman filter (which takes a step of the underlying state space model), but not taking the measurements into account in the update.
To this end I had to overwrite nfoursid `step()` function with a custom-made `step_accepting_missing_obs()` function. As this is not very future-proof, I've also fixed nfoursid's version in the requirements for the time being.

For more context about this choice, you can checkout https://github.com/spmvg/nfoursid/issues/3